doc/testing.md

```diff
@@ -0,0 +1,51 @@
+# Testing
+
+Seastar leverages Boost.Test and provides facilitiets for developers to implement tests in coroutines.
+
+## Tests Declaration
+
+There are three categories of boost-based tests in our system:
+
+* Boost.Test Native: Tests defined using `BOOST_AUTO_TEST_CASE` and related macros from Boost.Test. For more information, see the [Boost Test documentation](https://www.boost.org/doc/libs/release/libs/test/doc/html/boost_test/utf_reference/test_org_reference.html).
+* Coroutine: Tests defined using `SEASTAR_TEST_CASE`. The test body returns a future, allowing implementation as a coroutine.
+* Coroutine with `seastar::thread`: Tests defined using `SEASTAR_THREAD_TEST_CASE`. The test body is launched in a Seastar thread, allowing the use of Seastar coroutines. These tests should return `void`.
+
+## Choosing the Appropriate Macro
+
+* Use `SEASTAR_TEST_CASE` or `SEASTAR_THREAD_TEST_CASE` if you need to run tests in a Seastar reactor thread, typically for Seastar coroutines. `SEASTAR_THREAD_TEST_CASE` can be more convenient in some cases.
+* For Seastar facilities that don't depend on coroutines, consider using `BOOST_AUTO_TEST_CASE`.
+
+
+## Tests organization
```

Comment:
`Tests' Organization` or `Organizing Tests` look more correct IMO.

Reply:
i picked the former.